ui/src/shared/actions/annotations.ts

```diff
@@ -283,15 +296,61 @@ export const getAnnotationsAsync = (
 export const deleteAnnotationAsync = (
   annotation: Annotation
 ) => async dispatch => {
-  await api.deleteAnnotation(annotation)
-  dispatch(deleteAnnotation(annotation))
+  try {
+    dispatch(deleteAnnotation(annotation))
+    await api.deleteAnnotation(annotation)
+  } catch {
+    dispatch(notify(annotationsError('Error deleting annotation')))
+    dispatch(addAnnotation(annotation))
+  }
 }
 
 export const updateAnnotationAsync = (
   annotation: Annotation
 ) => async dispatch => {
-  await api.updateAnnotation(annotation)
-  dispatch(updateAnnotation(annotation))
+  try {
+    await api.updateAnnotation(annotation)
+    dispatch(updateAnnotation(annotation))
+  } catch {
+    dispatch(notify(annotationsError('Error saving annotation')))
```

Comment:
and if the update fails, should we update it back to its original state? previously this has been done by re-updating to whatever first came in to the function, and then the update on success would be an update with whatever is returned from the success. unlesss they're the same